interoperator/controllers/multiclusterdeploy/sfclusterreplicator/sfclusterreplicator_controller.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2019 The Service Fabrik Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sfclusterreplicator
+
+import (
+	"context"
+
+	resourcev1alpha1 "github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/api/resource/v1alpha1"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/controllers/multiclusterdeploy/watchmanager"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/cluster/registry"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/watches"
+
+	"github.com/go-logr/logr"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// To the function mock
+var getWatchChannel = watchmanager.GetWatchChannel
+
+// SFClusterReplicator replicates SFCluster
+type SFClusterReplicator struct {
+	client.Client
+	Log             logr.Logger
+	Scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+}
+
+// Reconcile reads that state of the cluster for a SFCluster object on master and sister clusters
+// and replicates it.
+func (r *SFClusterReplicator) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("sfcluster", req.NamespacedName)
+
+	cluster := &resourcev1alpha1.SFCluster{}
+	replica := &resourcev1alpha1.SFCluster{}
+	err := r.Get(ctx, req.NamespacedName, cluster)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	clusterID := cluster.GetName()
+	if clusterID == constants.OwnClusterID {
+		// Target cluster is mastercluster itself
+		// Replication not needed
+		return ctrl.Result{}, nil
+	}
+
+	targetClient, err := r.clusterRegistry.GetClient(clusterID)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	err = targetClient.Get(ctx, req.NamespacedName, replica)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.
+			// TODO: now wait for mcd provisioner to create. Fix me
```

Comment:
This TODO would not be required since the moment mcd creates the sfcluster, this one would get a trigger, right ?

Reply:
Comment if from Copy paste :(. Will remove